man/man8/zfs.8

```diff
@@ -31,316 +31,19 @@
 .\" Copyright 2019 Joyent, Inc.
 .\"
 .Dd June 30, 2019
-.Dt ZFS 8 SMM
-.Os Linux
+.Dt ZFS 8
+.Os
```

Comment:
I'm seeing that the old rendering (from zpool manpage) looks like:
```
$ man ./zpool.8
ZPOOL(8)                    System Manager's Manual                   ZPOOL(8)
...
Linux                           August 9, 2019                           Linux
```
And the new rendering:
```
$ man ./zfs.8
ZFS(8)                    BSD System Manager's Manual                   ZFS(8)
...
BSD                              June 30, 2019                             BSD
```

The change from `Linux` to `BSD` doesn't seem right to me, given that I'm running this on Ubuntu:
```
$ uname -a
Linux mahrens-lx.dc1 4.15.0-65-generic #74-Ubuntu SMP Tue Sep 17 17:06:04 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
```

FWIW, the existing (unmodified) zfs-program.8 manpage renders this way (with `BSD`) as well.

Reply:
The resolution is that OpenZFS will have to put the correct OS name in at build time. As we're going to have to add some preprocessing for platform-specific documentation, I'll revert these changes so the "default" OS is Linux and make them ifdef in a future PR.

#### Explanation
The behavior differs between `mandoc` from the "BSD" `mandoc` package (available on Ubuntu) and `man` from Ubuntu's `man-db` package, which calls `groff` to format the manpages:
```console
# man -Mman zfs.8 | head -n1
ZFS(8)                    BSD System Manager's Manual                   ZFS(8)
# mandoc man/man8/zfs.8 | head -n1
ZFS(8)                      System Manager's Manual                     ZFS(8)
```

Groff handles the `.Os` macro differently and wrongly. The problematic lines in the `groff-base` package are:
```
/usr/share/groff/1.22.4/tmac/mdoc/doc-common:333:.ds doc-default-operating-system BSD
/usr/share/groff/1.22.4/tmac/mdoc/doc-common:605:.ds doc-operating-system "\*[doc-default-operating-system]
```